goals.md

```diff
@@ -0,0 +1,61 @@
+As it has turned out unrealistic to unify/agglomerate these efforts into larger, and thus more resilient, projects, we have changed tack.
+We want to enable users to pick drivers from some instrument control libraries with a reasonably homogeneous/simple API, a nice GUI from another project, and a data storage solution from elsewhere, and tie them together to serve the needs of _their_ setup.
+
+[PyMeasure](https://pymeasure.readthedocs.io) is our project, so we know it best, and this will surely influence the design effort.
+Also, we plan a reference implementation in Python that will also probably be primarily tested with PyMeasure.
+However, we intend PyMeasure to be only one of _many_ "customer" libraries that can interface with ECP.
+
+This specification will be independent of the programming language its Components are written in, so it will be possible to mix-and-match.
+Instead of implementing all the necessary Components, we want to give the communication tools to connect everything, to balance on the shoulders of giants.
+
+## Small to medium-sized use cases
+We want ECP to serve laboratory setups of up to approximately 50 participants on one or more computation Nodes on a common network.
+For example, you might have up to ~dozens of instruments, a few data acquisition nodes, one or several GUIs and control nodes.
+
+## Flexible and modular
+"LEGO"-like modularity is important. We want to reassemble, change the constellation of devices, easily adapt to changing experiments, data rate needs, etc.
+
+We want people to easily be able to write their own software implementing (parts of) this protocol, for example to create just the GUI tool that they need.
+
+We want ECP to be very lightly coupled to the used software (like instrument driver libraries) -- we want to thinly wrap these with some interface code that is written (ideally) _once_.
+Given some instrument control library with a reasonably homogeneous/simple API, it should be easy to write an Actor variant to connect that library's Drivers with our Actor interface.
+
+## Simple, yet powerful
+You can send a message to any Component by specifying its (possibly human-readable) name or ID, similar to email.
+
+We will have a default interpretation of the message content to implement remote procedure call (RPC), i.e. to set/read properties or call methods.
+
+## Fast yet reliable
+We want to communicate asynchronously and concurrently: The Director might want to send several commands to different devices at the same time, without waiting for the answer of a (slow) instrument before writing to the next one.
+This way, communication delays will be determined by the slowest instrument, not by the number of instruments.
+
+Messages should be reliably delivered by the protocol and not lost (barring network problems, crashing Components, hardware failure, etc.).
+
+A failing node should not corrupt the measurement data acquired until then.
```

Comment:
Here again, the question arises, what is a node.
If a computer dies (e.g. hard drive), the data on that drive might be lost.

Do you mean a Component? That would mean, that a failing Actor does not corrupt data of the other Actors, and that a failing observer has to have written the data to permanent storage before failing (e.g. due to regular writing).

Reply:
Not sure actually, this came out of [your comment](https://github.com/pymeasure/experiment_control_protocol/issues/6#issuecomment-1370619410). 
I would say "Component", yes. I'll include your expanded reasoning, let us then revisit if this captures what you want to cover.

Reply:
In that other comment, I used the term "node" for a participant of ECP, which we now defined as Component

Reply:
Ok, all clear then.